### PR TITLE
Signup validation #22

### PR DESCRIPTION
**Pull request by BenNewman90**
Username, password and email validated with a relevant error message for each.

### CODE REVIEW

File: client/src/components/App/App.js [hunk 1]
```diff
@@ -21,19 +21,47 @@ class App extends Component {
...
 
   handleChange(event) {
-
     const { name } = event.target;
     const { value } = event.target;
     this.setState({ [name]: value });
+    let errorMessage = "";
+    let disabled = true;
+    if (
+      this.state.newUsername === "" &&
```
Comment (Karyum) [inline client/src/components/App/App.js]: also use `.trim()` on the strings so users can't input empty spaces as a username and `.trim()` just removes these empty spaces, and if the user only inputed empty spaces then `trim()` is going to trim it down to empty string

File: client/src/components/App/App.js [hunk 2]
```diff
@@ -21,19 +21,47 @@ class App extends Component {
...
+    ) {
+      errorMessage = "";
+    } else if (!this.state.newUsername.match(/[0-9a-zA-z]{6,20}/g)) {
+      errorMessage =
+        "Username must contain only numbers and letters with length between 6 and 20";
+    } else if (
+      !/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(this.state.newEmail)
+    ) {
+      errorMessage = "Must use a valid email address";
+    } else if (!this.state.newPassword.match(/[0-9a-zA-z]{6,20}/g)) {
```
Comment (Karyum) [inline client/src/components/App/App.js]: should we add like a minimum length for the password?

File: client/src/pages/SignupPage/SignupPage.js [hunk 3]
```diff
@@ -4,50 +4,56 @@ import Footer from "../../components/Footer/Footer.js";
 import register from "../../assets/images/register.png";
 import InputBox from "../../components/InputBox/InputBox.js";
 import Button from "../../components/Button/Button.js";
-import { SignupDiv } from "./SignupPage.style.js";
+import { SignupDiv, ErrorMsg } from "./SignupPage.style.js";
 
 class SignupPage extends Component {
   render() {
+    console.log(this.props);
```
Comment (Karyum) [inline client/src/pages/SignupPage/SignupPage.js]: we also don't need this log here

File: server/server.js [hunk 4]
```diff
@@ -38,8 +38,8 @@ app.get('/api/matches', (req, res) => {
     });
 });
 app.post('/api/login', (req, res) => {
-  console.log('im in the server');
-  res.body('success');
+  // console.log('im in the server');
+  // res.body('success');
   // getPassword(req.body.username)
   // .then(hash => bcrypt.compare(req.body.password, hash)
```
Comment (Karyum) [inline server/server.js]: i understand this is the signup validation PR that's why the login is commented out ? are you also going to add the sign up server logic in a different PR ?

File: client/src/components/App/App.js (hunk 2 above)
Comment (BenNewman90) [reply client/src/components/App/App.js]: The min is 6 and the max 20 '{6,20}'

**Pull request merged**